new_study_assistant.py

```diff
@@ -0,0 +1,624 @@
+
+        #with doc.create(Subsection('Number of common attributes')):
+            with doc.create(Figure(position='h!')) as n_common_attributes:
+                n_common_attributes.add_image(file_directory + 'n_common_attribute_distribution.png', width=NoEscape(r'0.99\textwidth'))
+                n_common_attributes.add_caption('Comparison between the number of common attributes in the new study and the number of common attributes in each existing study on cBioPortal.  The dashed black line indicates the number of common attributes in the new study, while the histogram shows the data for existing cBioPortal studies.')
+
+    doc.generate_pdf('report', clean_tex=True)
+
+def combine_matches(exact, name, value, no_matches):   
+    all_matches = (pd.concat([exact, value, name, no_matches])
+                      .drop_duplicates()
+                      .sort_values('New study attribute',axis=0)
+                      .reset_index(drop=True))
+    
+    for index, row in all_matches.iterrows():
+        append_string = ''
+        if (sum(exact['Possible matches'] == row['Possible matches']) > 0) or (sum(name['Possible matches'] == row['Possible matches']) > 0):
+            append_string += '^'
+        if sum(value['Possible matches'] == row['Possible matches']) > 0:
+            append_string += '*'
+        else:
+            append_string += ''
+
+        if row['Possible matches'] != 'No matches found':            
+            all_matches['Possible matches'][index] = all_matches['Possible matches'][index] + append_string + " (found in " + str(int(study_data_combined.sum()[row['Possible matches']])) + " other studies)"
+
+    return all_matches
+
+def attribute_type_prediction(exact_matches, name_matches, value_matches, studies):
+    sample_or_patient = {True:"Patient", False:"Sample"}
+    match_type_table = []
+    only_matches = (pd.concat([exact_matches, value_matches, name_matches])
+                      .drop_duplicates()
+                      .sort_values('New study attribute',axis=0)
+                      .reset_index(drop=True))
+    
+    for matching_attribute in only_matches['Possible matches'].values:
+        studies_with_attribute=study_data_combined[study_data_combined[matching_attribute]>0].index    
+        attribute_ps_types=[]
+        for study in studies_with_attribute:
+            study_index=study_names.index(study)
+
+            attribute_data=studies[study_index][1]
+            attribute_ps_type=attribute_data[attribute_data['clinicalAttributeId']==matching_attribute]['patientAttribute']
+            attribute_ps_types.append(sample_or_patient[attribute_ps_type.get_values()[0]])
+
+        match_type_table.append((matching_attribute, max(Counter(attribute_ps_types))))
+    return pd.DataFrame(match_type_table, columns=["Matching Atrribute","Patient or Sample prediction"])
+
+def generate_html_report(match_table, predicted_types, study_name):
```

Comment:
this parallels the generation of the latex / pdf output. I considered suggesting breaking down the report generation into functions to generate each section and taking an argument to determine the output format [latex, html, plaintext?] but I don't think that adds enough value to justify the effort. [it would codify standard report content and order across different formats, but you can achieve that manually - also you may have some sections that can not be rendered in certain formats, so it would not be perfectly symmetric anyway]